server/src/main/java/com/linecorp/centraldogma/server/internal/mirror/DefaultMirroringServicePlugin.java

```diff
@@ -48,27 +68,37 @@ public synchronized CompletionStage<Void> start(PluginContext context) {
         MirrorSchedulingService mirroringService = this.mirroringService;
         if (mirroringService == null) {
             final CentralDogmaConfig cfg = context.config();
-            final MirroringServicePluginConfig mirroringServicePluginConfig =
-                    (MirroringServicePluginConfig) cfg.pluginConfigMap().get(configType());
+            final MirroringServicePluginConfig mirroringServicePluginConfig = mirrorConfig(cfg);
             final int numThreads;
             final int maxNumFilesPerMirror;
             final long maxNumBytesPerMirror;
+            final ZoneConfig zoneConfig;
 
             if (mirroringServicePluginConfig != null) {
                 numThreads = mirroringServicePluginConfig.numMirroringThreads();
                 maxNumFilesPerMirror = mirroringServicePluginConfig.maxNumFilesPerMirror();
                 maxNumBytesPerMirror = mirroringServicePluginConfig.maxNumBytesPerMirror();
+                if (mirroringServicePluginConfig.zonePinned()) {
+                    zoneConfig = cfg.zone();
+                    if (zoneConfig == null) {
+                        throw new IllegalStateException(
```

Comment:
`zoneConfig` won't be null because it was checked in
https://github.com/line/centraldogma/blob/5f0a9a1eb7ed2db53db195cd5c8577d9bce8b459/server/src/main/java/com/linecorp/centraldogma/server/CentralDogma.java#L267-L270

Reply:
I just added this logic for double checking.

Reply:
Then, how about using `assert`? We use `assert` for this purpose. 😉

Reply:
Fixed.